### PR TITLE
[6-5] Coach history check table

### PR DESCRIPTION
**Pull request by GeranSkot**
1. Created start check test modal
2. Refactored check test page
3. Refactored data-table

### CODE REVIEW

File: exalearn-ui/src/app/components/data-table/data-table.component.html [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 		<ng-container matColumnDef="level">
 			<th mat-header-cell *matHeaderCellDef mat-sort-header>{{ 'TABLES.LEVEL' | translate}}</th>
-			<td mat-cell *matCellDef="let element">{{element.level}}</td>
+			<td mat-cell *matCellDef="let element">{{element.level || element.levelType}}</td>
```
Comment (KuznetsovR) [inline exalearn-ui/src/app/components/data-table/data-table.component.html]: Why did you add levelType here?

File: exalearn-ui/src/app/components/data-table/data-table.component.html [hunk 2]
```diff
@@ -34,7 +34,9 @@
 
 		<ng-container matColumnDef="expire">
 			<th mat-header-cell *matHeaderCellDef mat-sort-header>{{ 'TABLES.EXPIRE_DATE' | translate}}</th>
-			<td mat-cell *matCellDef="let element">{{element.expire || element.expireDate | date }}</td>
+			<td mat-cell *matCellDef="let element">
+				{{element.expire || element.expireDate || element.passedTestDate | date }}
```
Comment (KuznetsovR) [inline exalearn-ui/src/app/components/data-table/data-table.component.html]: Same question about date

File: exalearn-ui/src/app/pages/check-test-page/check-test-page.component.ts [hunk 3]
```diff
@@ -8,7 +11,19 @@ import { EnglishLevels } from '../../enums/enums';
...
+	isDataAvailable = false;
 	tableColumns = ['level', 'expire', 'check'];
+	data: CheckCoachTest;
+
+	constructor(private user: UserService, private apiService: ApiService) {}
+
+	ngOnInit() {
+		this.apiService.getRequest(API_PATH.GET_UNCHECKED_TEST).then((data) => {
+			data.forEach((element: any) => {
+				element.levelType = Object.keys(EnglishLevels)[element.levelType - 1];
```
Comment (KuznetsovR) [inline exalearn-ui/src/app/pages/check-test-page/check-test-page.component.ts]: Maybe you should name it level, not leveltype, so we didn't have same values with different names

File: exalearn-ui/src/app/components/check-essay/check-essay.component.scss [hunk 4]
```diff
@@ -1,38 +1,38 @@
...
 }
 
 @media screen and (max-width: 650px) {
-  .essay-container {
-    margin-top: 0;
-  }
+	.essay-container {
+		margin-top: 0;
+	}
 }
```
Comment (vstrabko) [inline exalearn-ui/src/app/components/check-essay/check-essay.component.scss:38]: What happened in this file?)
Incorrect tabulation?

File: exalearn-ui/src/app/services/submit-test.service.ts [hunk 5]
```diff
@@ -22,6 +22,7 @@ export default class SubmitTestService {
 	}
 
 	submitData(): void {
+		console.log(Array.from(this.mapOfAnswers.values()));
```
Comment (vstrabko) [inline exalearn-ui/src/app/services/submit-test.service.ts]: left out console.log here

File: exalearn-ui/src/assets/locale/ru_RU.json [hunk 6]
```diff
@@ -187,5 +187,10 @@
 		"DESCRIPTION": "Если вы думаете, что здесь нет правильных ответов или вы не можете ответить на этот вопрос по религиозным, политическим или иным причинам, вы можете пожаловаться на него и ваша оценка за тест будет изменена",
 		"TEXTAREA_LABEL": "Напишите комментарий",
 		"REPORT": "Пожаловаться"
+	},
+	"START_CHECK_MODAL": {
+		"START_CHECK_TEXT": "Вы действительно хотите начать проверять этот тест ?",
```
Comment (vstrabko) [inline exalearn-ui/src/assets/locale/ru_RU.json]: "начать проверку этого теста" sounds a bit better, doesn't it?

**Review by vstrabko — CHANGES_REQUESTED** [review]

**Review by vstrabko — APPROVED** [review]

**Review by KuznetsovR — APPROVED** [review]
